org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/ContentModelFormatterParticipant.java

```diff
@@ -67,4 +67,30 @@ public FormatElementCategory getFormatElementCategory(DOMElement element,
 		return null;
 	}
 
+	@Override
+	public boolean shouldCollapseEmptyElement(DOMElement element, SharedSettings sharedSettings) {
+		boolean enabled = sharedSettings.getFormattingSettings().isGrammarAwareFormatting();
+		if (!enabled) {
+			return true;
+		}
+		if (element.getAttribute("xsi:nil") != null) {

```

Comment:
I think the only usecase where schema must be search is when you have `xsi:nil=true"`

In other words, I will write that:

```java
if (!("true".equals(element.getAttribute("xsi:nil"))) {
    return true;
}
```